src/views/Diff.vue

```diff
@@ -0,0 +1,180 @@
+						'\n' +
+						'Mauris sed eros enim. Sed viverra semper nunc, et ornare tellus ullamcorper eu. Nullam quam nisl, posuere ac ipsum quis, vehicula sollicitudin dolor. Pellentesque hendrerit purus sed lacus euismod porta. Mauris aliquam consectetur sem nec imperdiet. Nam urna leo, rutrum at rutrum eget, euismod sed dui. Morbi sit amet libero eget urna rhoncus pellentesque eget condimentum eros. Aenean vehicula est quis dolor sodales scelerisque. Morbi imperdiet urna eget volutpat ornare. Phasellus feugiat leo eget lectus egestas gravida. Morbi hendrerit imperdiet enim at porttitor. Sed fermentum ac risus lacinia egestas. Donec malesuada velit nec quam commodo, in laoreet mauris tempor.',
+				},
+				matches: [
+					{ fromA: 5, toA: 150, fromB: 15, toB: 150 },
+					{ fromA: 290, toA: 400, fromB: 800, toB: 1000 },
+					{ fromA: 550, toA: 900, fromB: 220, toB: 600 },
+				],
+			});
+		},
+		highlightedText: function () {
+			const indices = this.documents.matches
+				.map((matches) => ({
+					fromA: matches.fromA,
+					toA: matches.toA,
+					fromB: matches.fromB,
+					toB: matches.toB,
+					color: this.hex_color_generator(),
+				}))
+				.flat();
+
+			const indicesA = indices.slice(0);
+			const indicesB = indices.slice(0);
+
+			indicesA.sort((a, b) => b.toA - b.fromA - (a.toA - a.fromA));
+			indicesB.sort((a, b) => b.toB - b.fromB - (a.toB - a.fromB));
+
+			const subStringToReplaceA = indicesA.map((h) => ({
+				text: this.documents.textA.obsah.substring(h.fromA, h.toA + 1),
+				color: h.color,
+			}));
+
+			const subStringToReplaceB = indicesB.map((h) => ({
+				text: this.documents.textB.obsah.substring(h.fromB, h.toB + 1),
+				color: h.color,
+			}));
+
+			this.doneA = this.merge_text(
+				subStringToReplaceA,
+				this.documents.textA.obsah
+			);
+
+			this.doneB = this.merge_text(
+				subStringToReplaceB,
+				this.documents.textB.obsah
+			);
+		},
+		hex_color_generator: function () {
+			return '#' + ((Math.random() * 0xffffff) << 0).toString(16);
+		},
+		merge_text(subStringIntervals, text) {
+			return subStringIntervals.reduce(
+				(string, substring) =>
+					string.replace(
+						new RegExp(substring.text, 'g'),
```

Comment:
This means it will globally replace some text. Is this really want we want? I thought we're replacing by som range.

Reply:
The text that is being replaced is still specific to that range. There is a small probability that you will have exactly two identical paragraphs somewhere.

Reply:
Makes sense.